lib/ansible/modules/remote_management/manageiq/manageiq_user.py

```diff
@@ -0,0 +1,288 @@
+
+        return dict(changed=True, msg=result['message'])
+
+    def edit_user(self, user, name, group, password, email):
+        """ Edit a user from manageiq.
+
+        Returns:
+            a short message describing the operation executed.
+        """
+        group_id = None
+        url = '{api_url}/users/{id}'.format(api_url=self.api_url, id=user['id'])
+
+        resource = dict(userid=user['userid'])
+        if group is not None:
+            group_id = self.group_id(group)
+            resource['group'] = dict(id=group_id)
+        if name is not None:
+            resource['name'] = name
+        if password is not None:
+            resource['password'] = password
+        if email is not None:
+            resource['email'] = email
+
+        # check if we need to update ( compare_user is true is no difference found )
+        if self.compare_user(user, name, group_id, password, email):
+            return dict(
+                changed=False,
+                msg="user {userid} is not changed.".format(userid=user['userid']))
+
+        # try to update user
+        try:
+            result = self.client.post(url, action='edit', resource=resource)
+        except Exception as e:
+            self.module.fail_json(msg="failed to update user {userid}: {error}".format(userid=user['userid'], error=e))
+
+        return dict(
+            changed=True,
+            msg="successfully updated the user {userid}: {user_details}".format(userid=user['userid'], user_details=result))
+
+    def create_user(self, userid, name, group, password, email):
+        """ Creates the user in manageiq.
+
+        Returns:
+            the created user id, name, created_on timestamp,
+            updated_on timestamp, userid and current_group_id.
+        """
+        # check for required arguments
+        for key, value in dict(name=name, group=group, password=password).items():
+            if value in (None, ''):
+                self.module.fail_json(msg="missing required argument: {}".format(key))
+
+        group_id = self.group_id(group)
+        url = '{api_url}/users'.format(api_url=self.api_url)
+
+        resource = {'userid': userid, 'name': name, 'password': password, 'group': {'id': group_id}}
+        if email is not None:
+            resource['email'] = email
+
+        # try to create a new user
+        try:
+            result = self.client.post(url, action='create', resource=resource)
+        except Exception as e:
+            self.module.fail_json(msg="failed to create user {userid}: {error}".format(userid=userid, error=e))
+
+        return dict(
+            changed=True,
+            msg="successfully created the user {userid}: {user_details}".format(userid=userid, user_details=result['results']))
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec=dict(
+            manageiq_connection=dict(required=True, type='dict'),
```

Comment:
requiring this as a dict and not using suboptions nor no_log makes the connection password too exposed.